ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngineClassLoader.java

```diff
@@ -120,6 +136,9 @@ public static <T, S, I extends Object> S initInstance(T type, I in, Class<?> con
         if (clazz == null) {
             clazz = connectorExecutorMap.get(type);
         }
```

Comment:
Just curious why is there 3 explicit checks for `clazz` being null?

Reply:
It needs to search three different Maps to find the instance.